src/t8_geometry/t8_geometry_implementations/t8_geometry_cad.hxx

```diff
@@ -362,16 +209,16 @@ struct t8_geometry_cad: public t8_geometry_with_vertices
   t8_geom_evaluate_cad_prism (t8_cmesh_t cmesh, t8_gloidx_t gtreeid, const double *ref_coords, const size_t num_coords,
                               double *out_coords) const;
 
-  const int *edges;                                /**< The linked edges of the currently active tree. */
-  const int *faces;                                /**< The linked faces of the currently active tree. */
-  TopoDS_Shape cad_shape;                          /**< cad geometry */
-  TopTools_IndexedMapOfShape cad_shape_vertex_map; /**< Map of all TopoDS_Vertex in shape. */
-  TopTools_IndexedMapOfShape cad_shape_edge_map;   /**< Map of all TopoDS_Edge in shape. */
-  TopTools_IndexedMapOfShape cad_shape_face_map;   /**< Map of all TopoDS_Face in shape. */
-  TopTools_IndexedDataMapOfShapeListOfShape
-    cad_shape_vertex2edge_map; /**< Maps all TopoDS_Vertex of shape to all its connected TopoDS_Edge */
-  TopTools_IndexedDataMapOfShapeListOfShape
-    cad_shape_edge2face_map; /**< Maps all TopoDS_Edge of shape to all its connected TopoDS_Face */
+  const int *edges; /**< The linked edges of the currently active tree. */
+  const int *faces; /**< The linked faces of the currently active tree. */
+
+  t8_cad *cad_manager; /**< The CAD manager of the geometry. */
```

Comment:
Can you change this pointer to a `std::shared_ptr`?
Then we can re-use the same object elsewhere in the code